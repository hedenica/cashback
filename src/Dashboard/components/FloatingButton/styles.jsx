import styled from 'styled-components';
import { shade } from 'polished';

export const Container = styled.button`
  width: 70px;
  height: 70px;
  border-radius: 50%;
  background: #fbab8c;
  color: #f7f7f7;
  padding: 10px;
  position: fixed;
  bottom: 5%;
  right: 5%;

  svg {
    width: 50px;
    height: 50px;
  }

  &:hover {
    background: ${shade(0.1, '#fbab8c')};
    box-shadow: 0px 3px 6px rgba(228, 178, 131, 0.4);
  }
`;
