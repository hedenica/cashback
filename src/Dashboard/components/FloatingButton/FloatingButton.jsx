import React from 'react';
import PropTypes from 'prop-types';
import { Plus } from 'react-feather';

import { Container } from './styles';

const FloatingButton = ({ showModal }) => (
  <Container type="button" onClick={showModal}>
    <Plus />
  </Container>
);

export default FloatingButton;

FloatingButton.propTypes = {
  showModal: PropTypes.func.isRequired,
};
