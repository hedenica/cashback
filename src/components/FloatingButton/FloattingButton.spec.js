import React from 'react';
import { render } from  '@testing-library/react'

import FloatingButton from './FloatingButton'

describe('<FloatingButton />', () => {
  it('should match the floating button snapshot', () => {
    const { container } = render(
      <FloatingButton showModal={() => true} />
    )  
    expect(container).toMatchSnapshot();
  });
})


