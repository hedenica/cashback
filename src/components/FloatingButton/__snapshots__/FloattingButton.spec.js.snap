// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FloatingButton /> should match the floating button snapshot 1`] = `
<div>
  <button
    class="sc-bdfBwQ dsDVol"
    type="button"
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="12"
        x2="12"
        y1="5"
        y2="19"
      />
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
    </svg>
  </button>
</div>
`;

exports[`it should match the floating button snapshot 1`] = `
<div>
  <button
    class="sc-bdfBwQ dsDVol"
    type="button"
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="12"
        x2="12"
        y1="5"
        y2="19"
      />
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
    </svg>
  </button>
</div>
`;
